executable-spec/src/Cardano/Ledger/Spec/STS/Chain/Chain.hs

```diff
@@ -113,105 +110,99 @@ instance ( HashAlgorithm hashAlgo
 
   type Environment (CHAIN hashAlgo) = Env hashAlgo
 
-  type State (CHAIN hashAlgo) = (St hashAlgo)
+  type State (CHAIN hashAlgo) = St hashAlgo
 
-  type Signal (CHAIN hashAlgo) = (Block hashAlgo)
+  type Signal (CHAIN hashAlgo) = Block hashAlgo
 
   data PredicateFailure (CHAIN hashAlgo)
-    = BlockSlotNotIncreasing CurrentSlot Slot
-    | MaximumBlockSizeExceeded Size (Threshold Size)
-    | TransactionsFailure (PredicateFailure (TRANSACTIONS hashAlgo))
+    = MaximumBlockSizeExceeded Size (Threshold Size)
+    | ChainFailureBody (PredicateFailure (BODY hashAlgo))
+    | ChainFailureHeader (PredicateFailure (HEADER hashAlgo))
     deriving (Eq, Show)
 
 
   initialRules = [
     do
-      IRC Env { initialSlot } <- judgmentContext
-      pure $! St { currentSlot = initialSlot
-                 , openVotingPeriods = Map.empty
-                 , closedVotingPeriods = Map.empty
-                 , transactionsSt = mempty
-                 }
+       IRC Env { headerEnv = Header.Env{ Header.initialSlot = initsl}
```

Comment:
Maybe we can define the initial rules of the chain in terms of the initial rules of the header and body STS's. I'm adding this comment here so that we can discuss it later.